src/framework/tooltip/Tooltip.jsx

```diff
@@ -43,11 +47,46 @@ export default class Tooltip extends Component {
       ) {
         tooltipY = -tooltipDimension.height - 19;
         styles.push('tooltipTop');
+        isPosYSet = true;
       }
 
-      this.container.className = styles.join(' ');
       this.container.style.left = `${tooltipX}px`;
       this.container.style.top = `${tooltipY}px`;
+
```

Comment:
You might wanna split it into two methods for better structuring and understanding why it has to calculated twice. Something like: 
```
componentDidUpdate() {
  if (this.state.hover) {
    calculatePositionToWindow()
    adjustPositionForVisibility()
  }
}
```

Also consider reusing calculations a little (e.g. see the updating the position to the right, basically `tooltipX`, setting style and class is the same).